### ISSUE TITLE
Protocol "https:" not supported

### ISSUE THREAD

**Issue by nilsbecker** [bitwarden/cli#181]
hi, i just installed bw but i'm unable to either check `bw update` or to `bw login`. i'm behind a http proxy which is set via environment variables and works for other CLI programs just fine.

with `bw login`, i get the error message in the title. googling it gave me this issue in `npm`: https://github.com/npm/cli/issues/2003 . it appears that this is an upstream bug in `https-proxy-agent`. they solved it by requiring a newer upstream version of `node-agent-base`. i think `bw` may be suffering from the same bug?

**Comment by kitos9112** [comment]
Hi,

I'm hitting the exact same issue here too.

**Comment by kitos9112** [comment]
Yes, @nilsbecker you're right it's an issue with that upstream library you mentioned earlier. I just built the CLI tool from source manually after changing the `package.json` file in the root directory to reflect the latest version of `https-proxy-agent` at 5.0.0 

Hencec, a simple `node ./build/bw.js login` worked magically as it should do by then asking for my `2FA code` behind my corporate proxy.

Not sure this re

**Comment by cscharf** [comment]
Hi @kitos9112 , sorry I had accepted/merged and overlooked those a little too quickly. Apparently the latest updated version of `https-proxy-agent` does break our build with a TypeScript compiler error, which we'll need to figure out before we can accept this change across the board. I've reverted these changes and am re-opening this issue here for further evaluation.

**Comment by kitos9112** [comment]
Hi @cscharf - That's perfectly fine, hope you can pinpoint the issue and correct it.

Thanks!

**Comment by kitos9112** [comment]
Hi @cscharf 

What's the state of your build server? should be re-open the already-closed MR and take a closer look at the codebase?

Many thanks!

**Comment by cscharf** [comment]
Hi @kitos9112 , we've not had an opportunity yet to dive into this, but it's on our radar.

**Comment by kitos9112** [comment]
Hello @cscharf - is there an update regarding this issue? It's kinda blocking for us to use the CLI tool internally inside our enterprise infrastructure where all outbound traffic needs to traverse an HTTP proxy.

Cheers

**Comment by xfoxfu** [comment]
To whom blocking on this issue, I have found a workaround for those who do not actually requires a proxy to connect to Bitwarden. Just unset all environment variables related to proxy, such as `HTTP_PROXY` to disable proxy. Here is a snippet:

```bash
unset HTTP_PROXY
unset HTTPS_PROXY
unset http_proxy
unset https_proxy
unset ALL_PROXY
unset all_proxy
```

**Comment by kitos9112** [comment]
Hi @xfoxfu your approach may work in an environment where you're either self-hosting your own BW appliance which is accessible via local routes (not reaching out to the Internet) or where despite having a proxy in place, your HTTP requests can still go through. None of those scenarios is my use case. I leverage the cloud offering of BW and all my outbound traffic is torn down before it leaves my default gateway.

I did notice @cscharf is working on fixing the build server on another repo which may be closely related --> https://github.com/bitwarden/jslib/pull/246

**Comment by eliykat** [comment]
Hi @kitos9112, this has been assigned to me and I'm looking into it. You're right that the action is happening over at -> https://github.com/bitwarden/jslib/pull/246.

**Comment by cscharf** [comment]
@eliykat , got that merged, if you could update jslib for cli that would be awesome, thanks!

**Comment by eliykat** [comment]
The upgrade to `https-proxy-agent` has now been merged, which according to @kitos9112 should solve the original issue here. I'll close this issue, but if the problem pops back up, please let us know.

**Comment by kitos9112** [comment]
@eliykat Thank you, however I cannot see a new PR opened to bump the external NPM dependency in this module.
@cscharf are we in good shape to re-open or cherry pick https://github.com/bitwarden/cli/pull/183 now?

**Comment by cscharf** [comment]
@eliykat , oh yeah, Thomas, can you please update in CLI and Directory Connector as well to mirror `jslib`? thanks!

**Comment by eliykat** [comment]
@kitos9112 Sorry about that - have now bumped it across all relevant repos:

* jslib (✅ )
* cli (✅ )
* directory-connector (✅) - but note this is waiting on an unrelated PR before it'll build properly, see the PR discussion if you're wanting to pull this in the meantime

Please ping me here if I've missed anything else or if it gives you any errors.
